xmlunit-assertj/src/main/java/org/xmlunit/assertj/ValidationAssert.java

```diff
@@ -0,0 +1,127 @@
+/*
+  This file is licensed to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package org.xmlunit.assertj;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.Assertions;
+import org.xmlunit.builder.Input;
+import org.xmlunit.validation.JAXPValidator;
+import org.xmlunit.validation.Languages;
+import org.xmlunit.validation.ValidationResult;
+
+import javax.xml.transform.Source;
+import javax.xml.validation.Schema;
+
+import static org.xmlunit.assertj.error.ShouldBeInvalid.shouldBeInvalid;
+import static org.xmlunit.assertj.error.ShouldBeValid.shouldBeValid;
+
+/**
+ * Assertion methods for XML validation.
+ *
+ * <p><b>Simple Example</b></p>
+ *
+ * <pre>
+ * import static org.xmlunit.assertj.XmlAssert.assertThat;
+ *
+ * final String xml = &quot;&lt;a&gt;&lt;b attr=\&quot;abc\&quot;&gt;&lt;/b&gt;&lt;/a&gt;&quot;;
+ *
+ * assertThat(xml).isValid();
+ * </pre>
+ *
+ * @since XMLUnit 2.6.1
+ */
+public class ValidationAssert extends AbstractAssert<ValidationAssert, Source> {
+
+    private final Source[] schemaSources;
+    private final Schema schema;
+
+    private ValidationAssert(Source actual, Source[] schemaSources, Schema schema) {
+        super(actual, ValidationAssert.class);
+        this.schemaSources = schemaSources;
+        this.schema = schema;
+    }
+
+    static ValidationAssert create(Object xmlSource, Object... schemaSources) {
+
+        Assertions.assertThat(xmlSource).isNotNull();
+
+        Assertions.assertThat(schemaSources)
+                .isNotNull()
+                .isNotEmpty()
```

Comment:
The Hamcrest Matcher allows for an empty list of sources which in the end leads to `JAXPValidator` using https://docs.oracle.com/javase/7/docs/api/javax/xml/validation/SchemaFactory.html#newSchema() - i.e. schema sources are picked up from the document's `xsi:schemaLocation` and `xsi:noSchemaLocation`.

Reply:
I assumed that clients whould know if schemaSources is empty or not and use `assertThat(xml).isValid()` or `assertThat(xml).isValidAgainst(schemaSources)` respectively.
But you have right I shouldn't expect that clients would check length of schemaSources